nx/lib/nx/constants.ex

```diff
@@ -277,19 +277,23 @@ defmodule Nx.Constants do
         0.0+1.0i
       >
 
-      iex> Nx.Constants.i(type: {:c, 128})
+      iex> Nx.Constants.i(:c128)
       #Nx.Tensor<
         c128
         0.0+1.0i
       >
 
   ## Error cases
 
-      iex> Nx.Constants.i(type: {:f, 32})
+      iex> Nx.Constants.i({:f, 32})
       ** (ArgumentError) invalid type for complex number. Expected {:c, 64} or {:c, 128}, got: {:f, 32}
   """
-  def i(opts \\ []) do
-    Nx.tensor(Complex.new(0, 1), opts)
+  def i(type \\ :c64, opts \\ []) do
+    if Keyword.has_key?(opts, :type) do
+      raise "type must not be passed as an option"
+    end
+
+    Nx.tensor(Complex.new(0, 1), Keyword.put(opts, :type, type))
   end
```

Comment:
I think it is just less guess work on what is the valid API and makes everything more consistent as the signatures all look like (type, opts \\ …) outside of the zero arity ones.

Reply:
What I mean is this:

```
iex(2)> h Nx.Constants.i

                        def i(type \\ :c64, opts \\ [])

Returns the imaginary constant.
```

the default type is already there as soon as you ask for the docs. If anything else, having a separate clause in this context makes it more work because you have to scroll a bit more to see that (and you have to actually read the text instead of just reading the default value)

Reply:
Yeah, the downside is that Nx.Constant.i(arg) is not clear what `arg` is. I am fine either way though, we should just be consistent.

Reply:
It can be confusing if it is type or opts. It is ambiguous unless you know exactly what it means. :)

Reply:
I see. I'll change it.